spring-core/src/main/java/org/springframework/core/metrics/micrometer/MicrometerStartupStep.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.core.metrics.micrometer;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+
+import org.springframework.core.metrics.StartupStep;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@link StartupStep} implementation for Spring Observability.
+ *
+ * @author Marcin Grzejszczak
+ */
+class MicrometerStartupStep implements StartupStep {
+
+	private final Timer.Sample sample;
+
+	private final String name;
+
+	private final Map<String, String> highCardinalityTags = new HashMap<>();
```

Comment:
Can this be a `List/Set<Tag>`?

We could do `highCardinalityTags.add(Tag.of(key, value));`
instead of `highCardinalityTags.put(key, value);`.

Also, converting this to `Tags` is trivial: `Tags.of(highCardinalityTags);`